src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher3.php

```diff
@@ -28,22 +28,45 @@ public function match($rawPathinfo)
             $canonicalMethod = 'GET';
         }
 
-        if (0 === strpos($pathinfo, '/rootprefix')) {
-            // static
-            if ('/rootprefix/test' === $pathinfo) {
+        switch ($pathinfo) {
+            case '/rootprefix/test':
+                // static
                 return array('_route' => 'static');
-            }
+                break;
+            case '/with-condition':
+                // with-condition
+                if (($context->getMethod() == "GET")) {
+                    return array('_route' => 'with-condition');
+                }
+                break;
+        }
 
-            // dynamic
-            if (preg_match('#^/rootprefix/(?P<var>[^/]++)$#sD', $pathinfo, $matches)) {
-                return $this->mergeDefaults(array_replace($matches, array('_route' => 'dynamic')), array ());
+        $regexOffset = 0;
+        $regexList = array(
+            '{^(?'
+            .'|/rootprefix/([^/]++)(*:_28)'
+            .')$}sD',
+        );
+
+        for ($i = 0; $i < \count($regexList); ++$i) {
+            if (!preg_match($regexList[$i], $pathinfo, $matches)) {
+                continue;
             }
 
-        }
+            switch ($m = $matches['MARK']) {
+                case '_28':
+                    $matches = array('var' => $matches[1] ?? null);
```

Comment:
In most routes the `$matches` var seems to be used only inside the `$this->mergeDefaults()` call. If that's always the case, maybe we can avoid this temporary variable and always include it inside `$this->mergeDefaults()`.

Reply:
these are dumped using independent methods, so cannot do it with heavy changes/complexity
since this is run-once code, there would be no practical benefit, so it's not useful